src/stability_sdk/animation.py

```diff
@@ -366,100 +434,183 @@ def render(self) -> Generator[Image.Image, None, None]:
+
+    def transform_2d(self, frame_idx) -> Optional[np.ndarray]:
+        if not len(self.prior_frames):
+            return None
+
+        args, frame_args = self.args, self.frame_args
+        angle = frame_args.angle_series[frame_idx]
+        scale = frame_args.zoom_series[frame_idx]
+        dx = frame_args.translation_x_series[frame_idx]
+        dy = frame_args.translation_y_series[frame_idx]
+
+        if angle == 0.0 and scale == 1.0 and dx == 0.0 and dy == 0.0:
+            return None
+
+        if args.accumulate_xforms:
+            frame_args = self.frame_args
+
+            # create xform for the current frame
+            xform = make_xform_2d(args.width, args.height, angle, scale, dx, dy)
+
+            # apply xform to prior frames running xforms
+            for i in range(len(self.prior_xforms)):
+                self.prior_xforms[i] = np.matmul(self.prior_xforms[i], xform)
+
+            # warp prior diffused frames by accumulated xforms
+            for i in range(len(self.prior_diffused)):
+                matrix = self.prior_xforms[i].copy().flatten().tolist()
+                op = generation.TransformOperation(
+                    warp2d=generation.TransformWarp2d(
+                        border_mode = border_mode_from_str_2d(args.border),
+                        matrix=generation.TransformMatrix(data=matrix)
+                ))                
+                xformed, _ = self.api.transform([self.prior_diffused[i]], [op])
+                self.prior_frames[i] = xformed[0]
+
+            return None
+        else:
+            op = warp2d_op(
+                border_mode=args.border,
+                rotate=angle, scale=scale, 
+                translate_x=dx, translate_y=dy, 
+            )
+            self.prior_frames, mask = self.api.transform(self.prior_frames, [op])
+            return mask
+
+    def transform_3d(self, frame_idx) -> Optional[np.ndarray]:
+        if not len(self.prior_frames):
+            return None
+
+        args, frame_args = self.args, self.frame_args
+        ops = [
+            depthcalc_op(blend_weight=args.midas_weight),
+            warp3d_op(
+                border_mode = args.border,
+                translate_x = frame_args.translation_x_series[frame_idx],
+                translate_y = frame_args.translation_y_series[frame_idx],
+                translate_z = frame_args.translation_z_series[frame_idx],
+                rotate_x = frame_args.rotation_x_series[frame_idx],
+                rotate_y = frame_args.rotation_y_series[frame_idx],
+                rotate_z = frame_args.rotation_z_series[frame_idx],
+                near_plane = args.near_plane,
+                far_plane = args.far_plane,
+                fov = frame_args.fov_series[frame_idx],
+            )
+        ]
+        self.prior_frames, mask = self.api.transform(self.prior_frames, ops)
+        return mask
+
+    def transform_video(self, frame_idx) -> Optional[np.ndarray]:
```

Comment:
we're not actually transforming the video here right? we're just estimating image transformation parameters *from* video, yeah? We might wanna rename this to clarify if that's correct.

Reply:
each mode (2D, 3D, and Video Input) has a corresponding transform_* method which handles updating prior frame state for the current animation frame. in the case of video it applies the optical flow as its transformation.

Reply:
would it maybe be clearer if we renamed these functions with an `_am_` prefix to denote that the suffx is specifying an **A**nimation **M**ode? So e.g. `transform_video()` -> `transform_am_video()`? it's a little awkward I guess, not sure what a good solution is here. I guess it's only really out of context in this PR and everywhere in the code it'll probably have the other transform funcs in proximity for that context. Maybe the only additional context we need here is a docstring